test/EFCore.PG.FunctionalTests/Scaffolding/NpgsqlDatabaseModelFactoryTest.cs

```diff
@@ -1413,6 +1413,48 @@ public void Index_operators()
                 },
                 @"DROP TABLE ""IndexOperators""");
 
+        [Fact]
+        public void Index_sort_order()
+            => Test(@"
+CREATE TABLE ""IndexSortOrder"" (a text, b text);
+CREATE INDEX ix_with ON ""IndexSortOrder"" (a DESC, b ASC);
+CREATE INDEX ix_without ON ""IndexSortOrder"" (a, b);",
+                Enumerable.Empty<string>(),
+                Enumerable.Empty<string>(),
+                dbModel =>
+                {
+                    var table = dbModel.Tables.Single();
+
+                    var indexWith = table.Indexes.Single(i => i.Name == "ix_with");
+                    Assert.Equal(new[] { SortOrder.Descending, SortOrder.Ascending }, indexWith.FindAnnotation(NpgsqlAnnotationNames.IndexSortOrder).Value);
+
+                    // Indexes default to ASC when no sort order is specified.
+                    var indexWithout = table.Indexes.Single(i => i.Name == "ix_without");
+                    Assert.Equal(new[] { SortOrder.Ascending, SortOrder.Ascending }, indexWithout.FindAnnotation(NpgsqlAnnotationNames.IndexSortOrder).Value);
```

Comment:
@roji Is there any point in trying to be smart here and not add the `NpgsqlAnnotationNames.IndexSortOrder` annotation if all sort orders are default values?

Reply:
That actually does seem like a good idea, this would save us some needless annotations in migration code, for example. It would be good if you can do this.

Reply:
> some needless annotations in migration code

Does this affect migration code though? Isn't this for code scaffolding? I.e. it would save some calls to `ForNpgsqlHasSortOrder` etc.?

Reply:
I'm thinking of doing something similar to this;

https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/blob/028316b8f467afde88228360912b6ba27e9ea36c/src/EFCore.PG/Scaffolding/Internal/NpgsqlDatabaseModelFactory.cs#L533-L539

Reply:
Maybe it's better to handle it in `IsHandledByConvention` in `NpgsqlAnnotationCodeGenerator` instead?

Reply:
Sorry, I mistakenly thought above that we were talking about the regular path, not scaffolding. But we definitely don't want to superfluous annotations when scaffolding for the default case, where they're no necessary - this would mean that every scaffolded index gets an unneeded `ForNpgsqlHasOrderSort()`, etc.

As to where to do this - NpgsqlDatabaseModelFactory or NpgsqlAnnotationCodeGenerator - I think it's simpler and easier to just do it in the model factory.

Reply:
Alright. Pushed an update. I think scaffolding should be done as well now. Hopefully that was the last piece of the puzzle 🤞